src/graph_API.jl

```diff
@@ -0,0 +1,90 @@
+type edge
+  # param_node: the "in" node (listed first in connectparameter)
+  # var_node: the "out" node (listed second in connectparameter)
+  name::Symbol
+  param_node::Symbol #name of component
+  var_node::Symbol #name of component
+end
+
+type ModelGraph
+  edges
+  nodes
+  size
+end
+ModelGraph(e,n) = ModelGraph(e, n, length(n))
+ModelGraph() = ModelGraph([],[],0)
+
+function add_edge(g::ModelGraph, e::edge)
+  @assert e.param_node in g.nodes && e.var_node in g.nodes ["Tried to add an edge between vertices not in the graph"]
+  append!(g.edges, [e])
```

Comment:
Why the ``[]`` around the ``e``? You want to add one element, not a vector of one element, right?